rnd/autogpt_server/autogpt_server/server/server.py

```diff
@@ -178,20 +182,63 @@ def run_service(self):
             methods=["POST"],
         )
 
-        app.add_exception_handler(500, self.handle_internal_error)  # type: ignore
+        app.add_exception_handler(500, self.handle_internal_error)
 
-        app.mount(
-            path="/frontend",
-            app=StaticFiles(directory=get_frontend_path(), html=True),
-            name="example_files",
-        )
+        class SPAStaticFiles(StaticFiles):
+            async def get_response(
+                self, path: str, scope: MutableMapping[str, Any]
+            ) -> Response:
+                try:
+                    return await super().get_response(path, scope)
+                except (HTTPException, StarletteHTTPException) as ex:
+                    if ex.status_code == 404:
+                        return await super().get_response("build.html", scope)
+                    else:
+                        raise ex
 
         app.include_router(router)
 
+        #! WARNING: THE ORDER OF THE REST OF THESE FUNCTIONS IS IMPORTANT
+        #! DO NOT CHANGE THE ORDER OF THESE FUNCTIONS
+        #! DO NOT ADD ANY FUNCTIONS BETWEEN THESE FUNCTIONS
+        #! DO NOT REMOVE ANY OF THESE FUNCTIONS
+        #! YOU WILL BREAK THE FRONTEND PACKAGED WITH THE SERVER
+
+        if os.path.exists(get_frontend_path() / "_next"):
+            app.mount(
+                path="/_next",
+                app=SPAStaticFiles(directory=get_frontend_path() / "_next", html=True),
+                name="frontend",
+            )
+
         @app.websocket("/ws")
         async def websocket_endpoint(websocket: WebSocket):  # type: ignore
             await self.websocket_router(websocket)
 
+        @app.get("/{extras_file:str}", response_class=HTMLResponse)
+        async def catch_others(extras_file: str):
+            # Do not allow directory traversal
+            if ".." in extras_file:
+                raise HTTPException(status_code=400, detail="Invalid file path")
+            if extras_file.count(".") > 1:
```

Comment:
If the file has no extension, can it use relative path ?
Can we use if not `os.path.isabs` to avoid the accessing relative path?